staging/src/k8s.io/kubectl/pkg/describe/describe.go

```diff
@@ -2381,7 +2397,89 @@ func (i *IngressDescriber) describeBackend(ns string, backend *networkingv1beta1
 	return formatEndpoints(endpoints, sets.NewString(spName))
 }
 
-func (i *IngressDescriber) describeIngress(ing *networkingv1beta1.Ingress, describerSettings DescriberSettings) (string, error) {
+func (i *IngressDescriber) describeBackendV1(ns string, backend *networkingv1.IngressBackend) string {
+	endpoints, err := i.client.CoreV1().Endpoints(ns).Get(context.TODO(), backend.Service.Name, metav1.GetOptions{})
+	if err != nil {
+		return fmt.Sprintf("<error: %v>", err)
+	}
+	service, err := i.client.CoreV1().Services(ns).Get(context.TODO(), backend.Service.Name, metav1.GetOptions{})
+	if err != nil {
+		return fmt.Sprintf("<error: %v>", err)
+	}
+	spName := ""
+	for i := range service.Spec.Ports {
+		sp := &service.Spec.Ports[i]
+		if backend.Service.Port.Number != 0 {
+			sNum := int64(backend.Service.Port.Number)
+			sp.Name = strconv.FormatInt(sNum, 10)
+		} else {
+			sp.Name = backend.Service.Port.Name
+		}
+	}
+	return formatEndpoints(endpoints, sets.NewString(spName))
+}
+
+func (i *IngressDescriber) describeIngressV1(ing *networkingv1.Ingress, events *corev1.EventList) (string, error) {
+	return tabbedString(func(out io.Writer) error {
+		w := NewPrefixWriter(out)
+		w.Write(LEVEL_0, "Name:\t%v\n", ing.Name)
+		w.Write(LEVEL_0, "Namespace:\t%v\n", ing.Namespace)
+		w.Write(LEVEL_0, "Address:\t%v\n", loadBalancerStatusStringer(ing.Status.LoadBalancer, true))
+		def := ing.Spec.DefaultBackend
+		ns := ing.Namespace
+		if def == nil {
+			// Ingresses that don't specify a default backend inherit the
+			// default backend in the kube-system namespace.
+			def = &networkingv1.IngressBackend{
+				Service: &networkingv1.IngressServiceBackend{
+					Name: "default-http-backend",
+					Port: networkingv1.ServiceBackendPort{
+						Number: 80,
+					},
+				},
+			}
+			ns = metav1.NamespaceSystem
+		}
+		if def.Service != nil {
```

Comment:
I would unnest this, make it apply to all backend types, and factor out the backend description to apply to both service and resource backends:

```
w.Write(LEVEL_0, "Default backend:\t%s\n", i.describeBackendV1(ns, def))
...
	w.Write(LEVEL_2, "\t%s \t%s\n", path.Path, i.describeBackendV1(ing.Namespace, &path.Backend))
```

inside describeBackendV1:
* for service backends, return the old `%s (%s)` format
* for resource backends, return `<apiGroup>.<kind> <name>` or something similar

Reply:
@liggitt Any chance you can run the describe test and spot where I'm making a mistake

```make test WHAT=./staging/src/k8s.io/kubectl/pkg/describe/...```

Reply:
NVM looks like my editor collapsed these return lines and I missed that